client/transaction-pool/src/graph/base_pool.rs

```diff
@@ -385,7 +385,7 @@ impl<Hash: hash::Hash + Member + Serialize, Ex: std::fmt::Debug> BasePool<Hash,
 	/// Removes and returns worst transactions from the queues and all transactions that depend on
 	/// them. Technically the worst transaction should be evaluated by computing the entire pending
 	/// set. We use a simplified approach to remove the transaction that occupies the pool for the
-	/// longest time.
+	/// longest time or has the lowest priority.
```

Comment:
Actually please update the comment, cause we now prefer transaction with the lowest priority and only resolve to `insertion_id` in case the priority is exactly the same.

Reply:
I can remember why it wasn't done like that initially, my only guess is that `self.ready.fold` did not traverse the entire set of transactions but rather just a subset (i.e. first-in-row transactions), but now it seems it will go through the list of all possible transactions anyway (which is a bit of a performance hurdle here, when used in the `while` loop).

Reply:
I had the same concern about performance, but wasn't sure if it is save to sort txs first and then slicing off some of the worst ones until we are withing limits (instead of having quadratic algorithm here) since `self.remove_subtree()` below might remove more than one by the looks of it and I wasn't familiar enough with the codebase to make such change yet.

Reply:
Well, in regular circumstances the outer loop should complete quite fast - like we will be removing up to couple of transactions at most, so I think this is fine.